### PR TITLE
A play to allow customers to set their VMware Tools repo

### PR DESCRIPTION
**Pull request by gungazoo**
The customer will be able to upload new versions of VMware Tools if they want to. Right now Tools is tied to the ESXi version.

### CODE REVIEW

**Review by et1975 — CHANGES_REQUESTED** [review]
Small nits.

File: Microsoft.AVS.Management/Microsoft.AVS.Management.psm1 [hunk 1]
```diff
@@ -1439,4 +1439,89 @@ function Set-HcxScaledCpuAndMemorySetting {
...
     }
-}
\ No newline at end of file
+}
+
+<#
+    .Synopsis
+     This will create a folder on every datastore (/vmfs/volumes/datastore/tools-repo) and set the ESXi hosts to use that folder as the tools-repo. The customer is responsible for putting the VMware Tools files into the folder.
+
+     .EXAMPLE
```
[the remark this answers is not in the document]
Comment (gungazoo) [reply Microsoft.AVS.Management/Microsoft.AVS.Management.psm1:1448]: I'm not sure. I just looked at it again. No trailing whitespace either.

**Review by gungazoo — COMMENTED** [review]
filter for VSAN

**Review by et1975 — APPROVED** [review]

File: Microsoft.AVS.Management/Microsoft.AVS.Management.psm1 [hunk 2]
```diff
@@ -1439,4 +1439,89 @@ function Set-HcxScaledCpuAndMemorySetting {
...
+    {
+        Write-Error -Message "Unable to get datastores. $($_.Exception.Message)"
+    }
+
+    foreach ($datastore in $datastores)
+    {
+        # Get datastore name
+        $ds_name = $datastore.Name
+
+        # Get ID of the vsanDatastore requrested
```
Comment (joesarabia) [inline Microsoft.AVS.Management/Microsoft.AVS.Management.psm1:1474]: nit: typo in comment

**Pull request merged**